lib/src/utils/format_tool_builder.dart

```diff
@@ -0,0 +1,234 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:dart_dev/src/utils/over_react_format_tool.dart';
+import 'package:glob/glob.dart';
+
+import '../../dart_dev.dart';
+import 'logging.dart';
+
+enum KnownFormatTools { FormatTool, OverReactFormatTool }
+
+/// Visits a `dart_dev/config.dart` file and searches for a custom formatter.
+///
+/// When a custom formatter is found, this will look for possible configuration options
+/// and reconstruct them on a new formatter instance.
+///
+/// NOTE: Because the visitor doesn't have access to the scope of the configuration
+/// being parsed, most values need to be a literal type (ListLiteral, StringLiteral, IntegerLiteral)
+/// to be reconstructed.
+///
+/// Expects the configuration to be in the format of:
+/// ```dart
+/// import 'package:over_react_format/dart_dev_tool.dart';
+///
+/// final config = {
+///   'format': OverReactFormatTool()
+///     ..lineLength = 120,
+/// };
+/// ```
+class FormatToolBuilder extends GeneralizingAstVisitor<void> {
+  DevTool formatDevTool;
+
+  FormatToolBuilder();
+
+  @override
+  visitMapLiteralEntry(MapLiteralEntry node) {
+    super.visitMapLiteralEntry(node);
+
+    String mapEntryKey = node.key.toSource().replaceAll("'", "");
+
+    if (mapEntryKey != 'format') return;
+
+    final formatterInvocation = node.value;
+    formatDevTool = detectFormatter(node.value);
+
+    if (formatterInvocation is CascadeExpression) {
+      List<Glob> reconstructedExcludesList = [];
+
+      final excludeExpression =
+          formatterInvocation.getCascadeByProperty('exclude');
```

Comment:
Do we need to reconstruct the excludes if we're using the rest args as the inputs instead of finding inputs like we normally would?

Reply:
We do not, I just thought while I was here I could add it so that `hackFastFormat` command works correctly without using the rest args in case a need ever existed. If it's detrimental to the goal of the ticket, I can remove it though!

Reply:
I'm still a little hesitant to even add this support for reconstructing excludes when we don't currently have a use case for it, especially given the code below and the possibility that warnings might be emitted that won't actually cause any issues in practice.

Reply:
👍  No problem! You're definitely right that it could cause issues and a more fully-baked implementation would only do this when necessary and not all the time. So I'll save the work as a spike in case it ever becomes important but can remove it from this PR!